trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -1353,6 +1430,84 @@ private Result handleAtomicExchangeN(final IDispatcher main, final IASTFunctionC
 				applyMemoryOrder(loc, atomicBuilder.build(), memoryOrder.getLrValue().getValue())).build();
 	}
 
+	// https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html#index-_005f_005fatomic_005fcompare_005fexchange_005fn
+	// https://en.cppreference.com/w/c/atomic/atomic_compare_exchange
+	//
+	// The implementation below generates Boogie code that roughly follows this schema, where success and ptr_val are
+	// auxiliary variables:
+	//
+	// @formatter:off
+	// (evaluate arguments)
+	// atomic {

```

Comment:
Is this block really atomic or does it depend on the memory order?

Reply:
I've added a comment explaining that we only follow this scheme if the memory order is sequential consistency, and otherwise `assert false`.